### PR TITLE
make master branch selenium 4 compliant

### PR DESCRIPTION
**Pull request by kamal-kaur04**
- Make master branch selenium 4 compliant

### CODE REVIEW

File: src/test/java/com/browserstack/BrowserStackTest.java [hunk 1]
```diff
@@ -36,44 +33,51 @@ public void setUp(String config_file, String environment) throws Exception {
 		JSONObject config = (JSONObject) parser.parse(new FileReader("src/test/resources/conf/" + config_file));
 		JSONObject envs = (JSONObject) config.get("environments");
 
-		DesiredCapabilities capabilities = new DesiredCapabilities();
-		capabilities.setCapability("browserstack.source","selenide:sample-master:v1.0");
+		MutableCapabilities capabilities = new MutableCapabilities();
 
 		Map<String, String> envCapabilities = (Map<String, String>) envs.get(environment);
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/BrowserStackTest.java]: Map<String, Object>

File: src/test/java/com/browserstack/BrowserStackTest.java [hunk 2]
```diff
@@ -36,44 +33,51 @@ public void setUp(String config_file, String environment) throws Exception {
...
 		Iterator it = envCapabilities.entrySet().iterator();
-    	while (it.hasNext()) {
-      		Map.Entry pair = (Map.Entry)it.next();
-      		capabilities.setCapability(pair.getKey().toString(), pair.getValue().toString());
-    	}
+		while (it.hasNext()) {
+			Map.Entry pair = (Map.Entry) it.next();
+			capabilities.setCapability(pair.getKey().toString(), pair.getValue());
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/BrowserStackTest.java:72]: Handling of bstack:options

File: src/test/java/com/browserstack/BrowserStackTest.java [hunk 3]
```diff
@@ -36,44 +33,51 @@ public void setUp(String config_file, String environment) throws Exception {
...
+		}
 
-		Map<String, String> commonCapabilities= (Map<String, String>) config.get("capabilities");
+		Map<String, String> commonCapabilities = (Map<String, String>) config.get("capabilities");
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/BrowserStackTest.java]: String , Object

File: src/test/java/com/browserstack/BrowserStackTest.java [hunk 4]
```diff
@@ -36,44 +33,51 @@ public void setUp(String config_file, String environment) throws Exception {
...
 		}
 
-		if (capabilities.getCapability("browserstack.local") != null && capabilities.getCapability("browserstack.local") == "true") {
-			l = new Local();
-			Map<String, String> options = new HashMap<String, String>();
-			options.put("key", accessKey);
-			l.start(options);
+		if (capabilities.getCapability("bstack:options") != null) {
+			HashMap bstackOptionsMap = (HashMap) capabilities.getCapability("bstack:options");
+			if ((bstackOptionsMap.get("local") != null && bstackOptionsMap.get("local").toString().equalsIgnoreCase("true")) || (capabilities.getCapability("browserstack.local") != null && capabilities.getCapability("browserstack.local").toString().equalsIgnoreCase("true"))) {
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/BrowserStackTest.java]: This is a sel4 sample and browserstack.local shouldn't be honoured

Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/BrowserStackTest.java]: Add check for ( l == null || ! l.isRunning() )

File: src/test/resources/conf/local.conf.json [hunk 5]
```diff
@@ -1,18 +1,20 @@
 {
-	"server": "hub.browserstack.com",
+	"server": "hub-cloud.browserstack.com",
```
Comment (karanshah-browserstack) [inline src/test/resources/conf/local.conf.json]: hub.browserstack.com

File: src/test/resources/conf/suite.conf.json [hunk 6]
```diff
@@ -1,23 +1,31 @@
...
     },
     "safari": {
-      "browser": "safari"
+      "bstack:options": {
+        "os": "OS X"
+      },
+      "browserName": "safari"
+    },
+    "ie": {
+      "browserName": "internet explorer"
```
Comment (karanshah-browserstack) [inline src/test/resources/conf/suite.conf.json]: IE could be removed from sample

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

**Review by karanshah-browserstack — APPROVED** [review]

**Review by francisf — APPROVED** [review]

**Pull request merged**